viz_scripts/plots.py

```diff
@@ -19,115 +20,87 @@
     v2l_df = pd.DataFrame({"vals": values}, index=labels)
 
     # Calculate % for all the values
     vs = v2l_df.vals.sum()
-    v2l_df["pct"] = v2l_df.vals.apply(lambda x: (x/vs) * 100)
-    disp.display(v2l_df)
-
-    # Find small chunks to combine
-    small_chunk = v2l_df.where(lambda x: x.pct <= 2).dropna()
-    misc_count = small_chunk.sum()
-
-    v2l_df = v2l_df.drop(small_chunk.index)
-    disp.display(v2l_df)
-
-    # This part if a bit tricky
-    # We could have already had a non-zero other, and it could be small or large
-    if "Other" not in v2l_df.index:
-        # zero other will end up with misc_count
-        v2l_df.loc["Other"] = misc_count
-    elif "Other" in small_chunk.index:
-        # non-zero small other will already be in misc_count
-        v2l_df.loc["Other"] = misc_count
-    else:
-        # non-zero large other, will not already be in misc_count
-        v2l_df.loc["Other"] = v2l_df.loc["Other"] + misc_count
-    disp.display(v2l_df)
-
-    return (v2l_df.index.to_list(),v2l_df.vals.to_list())
-
-
-def format_pct(pct, values):
-    total = sum(values)
-    absolute = int(round(pct*total/100.0))
-    return "{:.1f}%\n({:d})".format(pct, absolute) if pct > 4 else''
-
-
-def pie_chart_mode(plot_title,labels,values,file_name):
-    
-    colours = dict(zip(labels, plt.cm.tab20.colors[:len(labels)]))
-    fig, ax = plt.subplots(figsize=(10, 10), subplot_kw=dict(aspect="equal"))
-
-    m_labels, m_values = merge_small_entries(labels, values)
-    
-    wedges, texts, autotexts = ax.pie(m_values,
-                                      labels = m_labels,
-                                      colors=[colours[key] for key in labels],
-                                      pctdistance=0.75,
-                                      autopct= lambda pct: format_pct(pct, values),
-                                      textprops={'size': 23})
-
-    ax.set_title(plot_title, size=25)
-    plt.text(-1.3,-1.3,f"Last updated {arrow.get()}", fontsize=10)
-    plt.setp(autotexts, **{'color':'white', 'weight':'bold', 'fontsize':20})
-    plt.savefig(SAVE_DIR+file_name+".png", bbox_inches='tight')
-    plt.show()
+    v2l_df["pct"] = v2l_df.vals.apply(lambda x: round((x/vs) * 100, 1))
 
-def pie_chart_sensed_mode(plot_title,labels,values,file_name):
-    all_labels= ['IN_VEHICLE',
-                 'UNKNOWN',
-                 'WALKING',
-                 'AIR_OR_HSR',
-                 'BICYCLING',
-                 'OTHER']
+    return (v2l_df.index.to_list(),v2l_df.vals.to_list(), v2l_df.pct.to_list())
 
-    val2labeldf = pd.DataFrame({"labels": labels, "values": values})
+# Create dataframe with cols: 'Mode' 'Count' and 'Proportion'
+def process_trip_data(labels, values, trip_type):
+    m_labels, m_values, m_pct = calculate_pct(labels, values)
+    data_trip = {'Mode': m_labels, 'Count': m_values, 'Proportion': m_pct}
+    df_total_trip = pd.DataFrame(data_trip)
+    df_total_trip['Trip Type'] = trip_type
```

Comment:
Ditto for `trip_type`

Reply:
We are using `Trip Type` to pass into as mode_label into y in :  `bar = ax.barh(y=mode_label, width=mode_prop, height=bar_height, left=bar_width, label=mode, color=colors_combined[mode])`.
Would it be better to pass an extra argument directly inside `plot_stacked_bar_chart`?